src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -449,12 +449,12 @@ CreateOffer::bridged_cross (
             if (dry_offer (view, offers_leg1.tip ()))
             {
                 leg1_consumed = true;
-                have_bridge = (have_bridge && offers_leg1.step ());
+                have_bridge = (offers_leg1.step () && have_bridge);
             }
             if (dry_offer (view, offers_leg2.tip ()))
             {
                 leg2_consumed = true;
-                have_bridge = (have_bridge && offers_leg2.step ());
+                have_bridge = (offers_leg2.step () && have_bridge);
```

Comment:
Good catch on the need for an amendment!  I would've missed that.

My feeling is that if the code needs a comment, and can be rearranged to not need a comment, the rearrangement is a win.  Here's my shot:

```
if (view.rules().enabled(fixCrossStepping))
{
    leg1_consumed = dry_offer(view, offers_leg1.tip());
    if (leg1_consumed)
        offers_leg1.step();

    leg2_consumed = dry_offer(view, offers_leg2.tip());
    if (leg2_consumed)
        offers_leg2.step();

    have_bridge = leg1_consumed && leg2_consumed;
}
else
{
    if (dry_offer (view, offers_leg1.tip ()))
    {
        leg1_consumed = true;
        have_bridge = (have_bridge && offers_leg1.step ());
    }
    if (dry_offer (view, offers_leg2.tip ()))
    {
        leg2_consumed = true;
        have_bridge = (have_bridge && offers_leg2.step ());
    }
}
```

The bit I'm still uncomfortable about is this comment on line 361:

```
    // Note the subtle distinction here: self-offers encountered in the
    // bridge are taken, but self-offers encountered in the direct book
    // are not.
```

I'm not sure "self-offers" are correctly accounted for here because I'm not sure what a self-offer looks like.

Reply:
That’s already taken care of by the code elsewhere, so we’re all good. That comment may need to go away.

A self-offer is an offer where you cross against one of your own offers. In direct crossing, it’s safe to just ignore such offers (you’re only moving money from your left pocket to your right pocket) but in bridged offers you can’t, because it affects how much funds will flow.